what4/src/What4/Protocol/SMTLib2.hs

```diff
@@ -912,3 +937,128 @@ shutdownSolver _solver p = do
   ec <- Process.waitForProcess (solverHandle p)
   stopHandleReader (solverStderr p)
   return (ec,txt)
+
+
+-----------------------------------------------------------------
+-- Checking solver version bounds
+
+mkChunks :: [Word] -> [Versions.VChunk]
+mkChunks = map ((:[]) . Versions.Digits)
+
+-- | The minimum (inclusive) version bound for a given solver.
+--
+-- The keys come from @'smtWriterName'@ in @'WriterConn'@.
+-- See also https://github.com/GaloisInc/crucible/issues/194
+solverMinVersions :: Map String Version
+solverMinVersions =
+  [ -- TODO: Why is this verion required?
+    ( "Yices"
+    , Version { _vEpoch = Nothing, _vChunks = mkChunks [2, 6, 1], _vRel = []}
+    )
+  ]
+
+-- | The maximum (non-inclusive) version bound for a given solver.
+--
+-- The keys come from @'smtWriterName'@ in @'WriterConn'@.
+solverMaxVersions :: Map String Version
+solverMaxVersions = []
+
```

Comment:
We might also want a version blacklist, in addition to the min and max bounds...